core/src/main/java/org/mule/runtime/core/internal/profiling/AbstractProfilingService.java

```diff
@@ -0,0 +1,112 @@
+
+  @Inject
+  protected ServerNotificationManager notificationManager;
+
+  @Inject
+  private ServerNotificationHandler serverNotificationHandler;
+
+  FeatureFlaggingService featureFlags;
+
+  private final Set<NotificationListener<ProfilingNotification>> addedListeners = new HashSet<>();
+
+  @Override
+  public void initialise() throws InitialisationException {}
+
+  @Override
+  public void start() throws MuleException {
+    if (featureFlags.isEnabled(ENABLE_PROFILING_SERVICE)) {
+      registerNotificationListeners(getDiscoveryStrategy().discover());
+    }
+  }
+
+  private void registerNotificationListeners(Set<ProfilingDataConsumer<ProfilingEventContext>> profilerDataConsumers) {
+    profilerDataConsumers.forEach(this::registerNotificationListener);
+  }
+
+  private void registerNotificationListener(ProfilingDataConsumer<ProfilingEventContext> profilingDataConsumer) {
+    NotificationListener<ProfilingNotification> profilingNotificationListener =
+        new DefaultProfilingNotificationListener(profilingDataConsumer);
+    notificationManager.addListenerSubscription(profilingNotificationListener, pn -> filterByAction(profilingDataConsumer, pn));
+    addedListeners.add(profilingNotificationListener);
+  }
+
+  private boolean filterByAction(ProfilingDataConsumer<ProfilingEventContext> profilerDataConsumer,
+                                 ProfilingNotification profilerNotification) {
+    return profilerDataConsumer.profilerEventTypes().stream()
+        .anyMatch(
+                  eventType -> eventType.getProfilingEventName()
+                      .equals(profilerNotification.getActionName()))
+        &&
+        profilerDataConsumer.selector().test(((ProfilingEventContext) profilerNotification.getSource()));
+  }
+
+  @Override
+  public void dispose() {
+    if (!notificationManager.isDisposed()) {
+      addedListeners.forEach(listener -> notificationManager.removeListener(listener));
+    }
+  }
+
+  protected abstract ProfilerDataConsumerDiscoveryStrategy getDiscoveryStrategy();
+
+  public <T extends ProfilingEventContext> void notifyEvent(T profilerEventContext, ProfilingEventType<T> action) {
+    serverNotificationHandler.fireNotification(new ProfilingNotification(profilerEventContext, action));
+  }
+
+  @Inject
+  public void setFeatureFlags(FeatureFlaggingService featureFlags) {
+    this.featureFlags = featureFlags;
+  }
+
+  /**
+   * Configures {@link FeatureFlaggingService} for MULE-19588.
+   *
+   * @since 4.4
+   */
+  public static void configureEnableProfilingService() {
+    FeatureFlaggingRegistry featureFlaggingRegistry = FeatureFlaggingRegistry.getInstance();
+    featureFlaggingRegistry.registerFeatureFlag(ENABLE_PROFILING_SERVICE,
+                                                featureContext -> featureContext.getArtifactMinMuleVersion()
+                                                    .filter(muleVersion -> muleVersion.atLeast("4.5.0"))
```

Comment:
is this `4.5.0` the same as the one declared here: https://github.com/mulesoft/mule-api/pull/838/files#diff-9018e4b400b25f6397972e30bf9832135f6557d83869174a7c8a99e87f20f32aR150 ?

Reply:
Yes. I use it now.